packages/dev/serializers/src/glTF/2.0/glTFAnimation.ts

```diff
@@ -943,7 +956,7 @@ export class _GLTFAnimation {
     }
 
     private static _GetBasePositionRotationOrScale(
-        babylonTransformNode: TransformNode,
+        babylonTransformNode: any,

```

Comment:
Why is that any now?

Reply:
Because we access specific `Position` and `Rotation`, which are not available at `Node` level. And because it is private i do not see any problem to put `any` instead of `Node`. Maybe i miss some rules :-)

Reply:
It looks like this function is guaranteed to take a TransformNode. https://github.com/BabylonJS/Babylon.js/blob/a18bc8d066805484845d9c8e784d08f88588af3b/packages/dev/serializers/src/glTF/2.0/glTFAnimation.ts#L836
TypeScript automatically narrows the type when using `instanceof`, so you don't actually need to change the types for this function or `_ConvertFactorToVector3OrQuaternion`.

Reply:
right but also used line #783 which is the reason of the change. and #1043 for `_ConvertFactorToVector3OrQuaternion`

Reply:
All of these should use TransformNode right? If it's not a TransformNode, this function won't work or am I misunderstanding that?

Reply:
It's not so obvious. Camera and TransformNode share similar properties, such rotationQuaternion or Position, and this is what we use to animate both.
This is come from the fact, At the very begining of the process, the serializer use the Camera as a TransformNode to leverage this similarities.
`const nodes: Node[] = [...babylonScene.transformNodes, ...babylonScene.meshes, ...babylonScene.lights, ...babylonScene.cameras];`
So it's not a garanty to get a TransformNode on each of this function.
However, youre right, we might verify if the property is defined prior to use it.

Reply:
Ok, can these functions take a union type like `TransformNode | Camera`? That should avoid using `any`?